apm-agent-core/src/main/java/co/elastic/apm/agent/impl/context/ServiceTarget.java

```diff
@@ -154,13 +142,15 @@ public CharSequence getDestinationResource() {
             return name;
         }
 
-        if (destinationResource.length() == 0) {
-            if (type != null && type.length() > 0) {
-                destinationResource.append(type);
-                destinationResource.append("/");
-            }
-            destinationResource.append(name);
+        // will allocate a bit, but it's fine as it's only expected to be called once
+        // - when the span is dropped for dropped spans stats
+        // - when the span is serialized
+        StringBuilder destinationResource = new StringBuilder();
```

Comment:
Why not keep it as a final private field and do `destinationResource.setLength(0)` here?